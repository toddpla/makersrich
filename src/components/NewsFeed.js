import React, { Component } from 'react';
import { connect } from 'react-redux'
class NewsFeed extends Component {

  render() {
    return (
      <div className="newsfeed-assemble" >
        <div className="newsfeed-message">{this.props.newsfeed[0]}</div>
        <div className="newsfeed-message">{this.props.newsfeed[1]}</div>
        <div className="newsfeed-message">{this.props.newsfeed[2]}</div>
      </div>
    );
  }

}

const mapStateToProps = (state) => ({
  newsfeed: state.newsfeed
})

export default connect(mapStateToProps)(NewsFeed)
