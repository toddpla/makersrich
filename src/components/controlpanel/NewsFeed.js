import React, { Component } from 'react';
import { connect } from 'react-redux'
import database from '../../firebase/firebase'

class NewsFeed extends Component {

  state = {
    newsfeed: []
  }

  componentDidMount(){
    database.ref('newsfeed')
      .on('child_added', snap => {
      this.setState((prevState) => {
        if (prevState.newsfeed.length > 3) prevState.newsfeed.shift()
        prevState.newsfeed.push(snap.val())
        return {
          newsfeed: prevState.newsfeed
        }
      })
    })
  }

  render() {
    return (
      <div id="newsfeed-assemble" >
      <h1>Newsfeed</h1>
        {this.state.newsfeed.map((item, i) => {
          return (
            <div key={i} className="newsfeed-message">
              <strong>{item.displayName && item.displayName.split(" ")[0]}</strong> {item.message}
            </div>
          )
        })}
      </div>
    );
  }
}

export default NewsFeed
