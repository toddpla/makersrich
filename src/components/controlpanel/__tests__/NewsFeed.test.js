import React from 'react'
import NewsFeed from '../NewsFeed'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

test('renders correctly', () => {
  const wrapper = renderer.create(<NewsFeed />)
  expect(wrapper).toMatchSnapshot()
});
