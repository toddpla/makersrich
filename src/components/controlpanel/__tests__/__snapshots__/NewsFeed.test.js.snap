// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="newsfeed-assemble"
>
  <h1>
    Newsfeed
  </h1>
</div>
`;
